client/src/pages/ContractorSignUp.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from "react";
+
+function ContractorSignUp() {
+	const [contractorDetails, setContractorDetails] = useState({
+		userType: "contractor",
+		firstName: "",
+		lastName: "",
+		email: "",
+		company: "",
+		address: "",
+	});
+	const [emailError, setEmailError] = useState("");
+	// This variable shows success of the request to during sign up that comes back from server
+	const [resgisterStatus, setRegisterStatus] = useState("");
+
+	const validateEmail = (email) => {
+		const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+		return emailRegex.test(email);
+	};
+
+	async function postContractorDetails(userData) {
+		try {
+			const response = await fetch("/mock-api", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(userData),
+			});
+			const data = response.json();
+			if (!response.ok) {
+				setRegisterStatus(data.message);
+			}
+			setRegisterStatus(data.message);
+		} catch (error) {
+			setRegisterStatus(error.message);
+		}
+	}
+
+	const handleInputChange = (e) => {
+		const { name, value } = e.target;
+		setContractorDetails((prevDetails) => ({
+			...prevDetails,
+			[name]: value,
+		}));
+	};
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		if (validateEmail(contractorDetails.email)) {
+			setEmailError("");
+			postContractorDetails(contractorDetails);
+		} else {
+			if (!validateEmail(contractorDetails.email)) {
```

Comment:
You don't need to validate again, by being in the `else` branch, you already know the email is not valid